data_collator.py

```diff
@@ -0,0 +1,114 @@
+import math
+from dataclasses import dataclass
+from typing import Any, Callable, Dict, List, NewType, Optional, Tuple, Union
+
+
+import jax.numpy as jnp
+import numpy as np
+from numpy.random import poisson, permutation
+
+from transformers.tokenization_utils_base import BatchEncoding, PreTrainedTokenizerBase
+from transformers.data.data_collator import _collate_batch
+
+@dataclass
+class DataCollatorForTextInfilling:
+    tokenizer: PreTrainedTokenizerBase
+    mlm_probability: float = 0.15
+    poisson_lambda: float = 3.0
+    pad_to_multiple_of: Optional[int] = None
+
+    def __post_init__(self):
+        if self.tokenizer.mask_token is None:
+            raise ValueError
+
+    def __call__(self, examples: List[Union[List[int], jnp.ndarray, Dict[str, jnp.ndarray]]]
+                 ) -> Dict[str, jnp.ndarray]:
+        # Handle dict or lists with proper padding and conversion to tensor.
+        if isinstance(examples[0], (dict, BatchEncoding)):
+            batch = self.tokenizer.pad(examples, return_tensors="jax", pad_to_multiple_of=self.pad_to_multiple_of)
+        else:
+            batch = {"input_ids": _collate_batch(examples, self.tokenizer, pad_to_multiple_of=self.pad_to_multiple_of)}
```

Comment:
We also need to return the original input for the `decoder_input_ids` for the decoder input, maybe rename like so:

```
batch = {"decoder_input_ids": _collate_batch(examples, self.tokenizer, pad_to_multiple_of=self.pad_to_multiple_of)}
```

And then pass `decoder_input_ids` to `self.mask_tokens` below

That way `batch` will include `input_ids`, `decoder_input_ids` and `labels`

Reply:
Adding this now

Reply:
What do you mean by
> And then pass ```decoder_input_ids``` to ```self.mask_tokens below```
?